src/lib/index.js

```diff
@@ -25,14 +26,40 @@ function getDisplayName(WrappedComponent) {
   return WrappedComponent.displayName || WrappedComponent.name || 'Component';
 }
 
+function capitalize(s) {
+  return s.charAt(0).toUpperCase() + s.substring(1);
+}
+
 const TOO_LIGHT_FACTOR = 0.8;
 export function tooLight(color) {
   const hslColor = tinyColor(color).toHsl();
   return hslColor.l > TOO_LIGHT_FACTOR;
 }
 
+function getPlotIcons(plotly) {
+  const icons = Object.keys(plotly.PlotSchema.get().traces).reduce(
```

Comment:
we already pass the plotSchema down in the context so no need to call `.get()` here. See the editor.

re: your comment about computing this once... that seems reasonable.

Reply:
wait a sec though

Reply:
Do we need to iterate over all the traces? Presumably when we need to show an icon we already have the trace type.

Reply:
I would rather not store an object if we don't need it.